### ISSUE TITLE
postgresql loop

### ISSUE THREAD

**Issue by jasonvriends** [codemation/easyauth#25]
Following the Server instructions for postgres, I encounter the following endless loop.

Not sure where the 8091 comes into play?
Cannot connect to host 127.0.0.1:8091 ssl:False [Connection refused]

======

INFO:     Uvicorn running on http://0.0.0.0:8330 (Press CTRL+C to quit)
INFO:     Started reloader process [18384] using statreload
get_chart name: DefaultChartName
INFO:     Started server process [18386]
INFO:     Waiting for application startup.
07-04 12:21 EasyAuthServer WARNING  loading env vars from easyauth.json

[2021-07-04 12:21:23 -0400] [18394] [INFO] Starting gunicorn 20.1.0
[2021-07-04 12:21:23 -0400] [18394] [INFO] Listening at: http://127.0.0.1:8091 (18394)
[2021-07-04 12:21:23 -0400] [18394] [INFO] Using worker: uvicorn.workers.UvicornWorker
[2021-07-04 12:21:23 -0400] [18396] [INFO] Booting worker with pid: 18396
[2021-07-04 12:21:24 -0400] [18396] [INFO] Started server process [18396]
[2021-07-04 12:21:24 -0400] [18396] [INFO] Waiting for application startup.
07-04 12:21 EasyRpc-server /ws/easyauth ERROR    error running query: 
        create or replace function show_tables() returns SETOF text as $$
        SELECT
            table_name
        FROM
            information_schema.tables
        WHERE
            table_type = 'BASE TABLE'
        AND
            table_schema NOT IN ('pg_catalog', 'information_schema');
        $$
        language sql; 
    
Traceback (most recent call last):
  File "/home/python/venv/lib/python3.8/site-packages/aiopyql/data.py", line 393, in __process_queue
    results = await self.process_query_no_commit(self, conn, query_id, query)
  File "/home/python/venv/lib/python3.8/site-packages/aiopyql/postgres_connector.py", line 260, in process_query_no_commit
    results = await conn.fetch(query)
  File "/home/python/venv/lib/python3.8/site-packages/asyncpg/connection.py", line 583, in fetch
    return await self._execute(
  File "/home/python/venv/lib/python3.8/site-packages/asyncpg/connection.py", line 1625, in _execute
    result, _ = await self.__execute(
  File "/home/python/venv/lib/python3.8/site-packages/asyncpg/connection.py", line 1650, in __execute
    return await self._do_execute(
  File "/home/python/venv/lib/python3.8/site-packages/asyncpg/connection.py", line 1697, in _do_execute
    result = await executor(stmt, None)
  File "asyncpg/protocol/protocol.pyx", line 201, in bind_execute
asyncpg.exceptions.InvalidFunctionDefinitionError: return type mismatch in function declared to return text
DETAIL:  Actual return type is information_schema.sql_identifier.
[2021-07-04 12:21:24 -0400] [18396] [ERROR] Traceback (most recent call last):
  File "/home/python/venv/lib/python3.8/site-packages/starlette/routing.py", line 540, in lifespan
    async for item in self.lifespan_context(app):
  File "/home/python/venv/lib/python3.8/site-packages/starlette/routing.py", line 481, in default_lifespan
    await self.startup()
  File "/home/python/venv/lib/python3.8/site-packages/starlette/routing.py", line 516, in startup
    await handler()
  File "/home/python/venv/lib/python3.8/site-packages/easyauth/proxy.py", line 67, in db_setup
    db = await Database.create(
  File "/home/python/venv/lib/python3.8/site-packages/aiopyql/data.py", line 62, in create
    await db.load_tables(db)
  File "/home/python/venv/lib/python3.8/site-packages/aiopyql/postgres_connector.py", line 105, in load_tables
    await db.get(show_tables_create)
  File "/home/python/venv/lib/python3.8/site-packages/aiopyql/data.py", line 466, in get
    result = await self.execute(query, commit=False)
  File "/home/python/venv/lib/python3.8/site-packages/aiopyql/data.py", line 439, in execute
    raise result
  File "/home/python/venv/lib/python3.8/site-packages/aiopyql/data.py", line 393, in __process_queue
    results = await self.process_query_no_commit(self, conn, query_id, query)
  File "/home/python/venv/lib/python3.8/site-packages/aiopyql/postgres_connector.py", line 260, in process_query_no_commit
    results = await conn.fetch(query)
  File "/home/python/venv/lib/python3.8/site-packages/asyncpg/connection.py", line 583, in fetch
    return await self._execute(
  File "/home/python/venv/lib/python3.8/site-packages/asyncpg/connection.py", line 1625, in _execute
    result, _ = await self.__execute(
  File "/home/python/venv/lib/python3.8/site-packages/asyncpg/connection.py", line 1650, in __execute
    return await self._do_execute(
  File "/home/python/venv/lib/python3.8/site-packages/asyncpg/connection.py", line 1697, in _do_execute
    result = await executor(stmt, None)
  File "asyncpg/protocol/protocol.pyx", line 201, in bind_execute
asyncpg.exceptions.InvalidFunctionDefinitionError: return type mismatch in function declared to return text
DETAIL:  Actual return type is information_schema.sql_identifier.

**Comment by codemation** [comment]
@jasonvriends - thanks for reporting 

As I mentioned in my earlier mail, the most likely cause of this issue is a non easyauth table within the postgres database which cannot be parsed by aiopyql. Could you confirm if this was an empty database, or containing tables for other applications.

**Comment by codemation** [comment]
@jasonvriends 

I managed to reproduce the same issue on Postgres 12, but 11 and 13 are fine( if you clear DB files before starting). So seems to be an issue in discovery function setup in version 12 within aiopyql. I'll link a separate issue in aiopyql and investigate when I have some time. Can you try using version 11 or 13?

**Comment by codemation** [comment]
Issue will be investigated in aiopyql [issue](https://github.com/codemation/aiopyql/issues/5), if you do not have other questions, please close when you get a chance.
